docs/frc/3-Closed_Loop_Control/5-Motion_Profiling.md

```diff
@@ -0,0 +1,275 @@
+
+We don't actually have to worry about generating these profiles alone, since WPILib provides the `TrapezoidProfile` class to handle all of these calculations for us. However, I'm going to take a dive into deriving these physics below. Feel free to skip it if you're not interested.
+
+### Physics of Generating Trapezoidal Profiles
+
+We are given the following constants when trying to generate a trapezoidal motion profile, acceleration $a$, max velocity $v_{max}$, and distance $d$. To create the motion profile, we have to use the kinematic equations, which are listed below.
+
+$$v = v_0 + at$$
+$$\Delta x = \left(\frac{v + v_0}{2}\right)t$$
+$$\Delta x = v_0 t + \frac{1}{2}at^2$$
+$$v^2 = v_0^2 + 2a \Delta x$$
+
+First of all, we have to determine if we are going to be using a triangular profile or a trapezoidal profile. This means that we need to determine whether or not we will even be able to accelerate to the maximum velocity and then deccelerate to a stop before going the distance $d$. To do so, we can use the kinematic equations to determine the distance required to accelerate to the velocity $v_{max}$ from the initial velocity $v_0 = 0$.
+
+$$v_{max}^2 = 0^2 + 2ax \rightarrow x = \frac{v_{max}^2}{2a}$$
+
+Now we have two cases: either $x \leq \frac{d}{2}$ or $x > \frac{d}{2}$. For the first case, this means that when accelerating to the max speed and then deccelerating, we will actually go **farther** than our desired distance, which is bad. This means that we should **not** go all of the way to our max speed, but instead go to some intermediate speed. This is known as a triangular profile, and can be done by determining the length of time we need to spend accelerating to go halfway to our destination. Then, the same time will be spent deccelerating.
+
+$$\frac{d}{2} = 0 * t + \frac{1}{2}at_a^2 \rightarrow t_a = t_d = \sqrt{d / a}$$
+
+Now, we know how long this triangular profile will last during each segment, and we just need to know what the desired velocity is during each time $t$. For the first half of our motion when $0 \leq t \leq t_a$, we have that $$v = at$$
+
+For the second half of our motion when $t_a \lt t \leq t_a + t_d$, we have that $$v = at_a - a(t - t_a) = 2at_a - at$$.
+
+This concludes the triangular profile, and now we can go onto the trapezoidal profile. For the trapezoidal profile, we know that we can safely accelerate to our maximum velocity, cruise at that velocity for some time, and then deccelerate. To calculate how long it will take to accelerate and deccelerate, we can use the kinematic equations again.
+
+$$v_max = 0 + at_a \rightarrow t_a = t_d = \frac{v_max}{a}$$
+
+We also need to calculate the distance that we travelled during this time, which can be done with the following equation:
+
+$$v_max^2 = 0^2 + 2ax \rightarrow x = \frac{v_max^2}{2a}$$
+
+We then know that we will need to cruise at our max velocity for the remainder of the distance, which means that we will need to cruise for the time
```

Comment:
There's a small issue here with the LaTeX, where only the letter m of "max" is written in subscript